libpod/container_internal.go

```diff
@@ -976,7 +976,7 @@ func (c *Container) checkDependenciesRunning() ([]string, error) {
 		if err != nil {
 			return nil, errors.Wrapf(err, "error retrieving state of dependency %s of container %s", dep, c.ID())
 		}
-		if state != define.ContainerStateRunning {
+		if state != define.ContainerStateRunning && !strings.Contains(depCtr.Name(), "infra") {
```

Comment:
This isn't safe - need to check container configuration to see if it's an infra container.